nixos/modules/services/misc/gollum.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gollum;
+in
+
+{
+  options.services.gollum = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Enable the Gollum service.";
+    };
+
+    package = mkOption {
+      default = pkgs.gollum;
+      defaultText = "pkgs.gollum";
+      description = "Gollum package used for the service";
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = "IP address on which the web server will listen.";
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 4567;
+      description = "Port on which the web server will run.";
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
```

Comment:
Consider using `types.lines`, it has more sensible merge semantics for an option like this.

Reply:
That makes sense. Updated the PR.